tests/bluetooth/mesh_shell_util/src/main.c

```diff
@@ -0,0 +1,236 @@
+		if (elems[i].is_valid) {
+			zassert_true(!bt_mesh_str2ul(elems[i].str, &test_val),
+				     "Failed to convert: \"%s\"", elems[i].str);
+			zassert_true(test_val == elems[i].exp_val.ul_val, "%d != %d", test_val,
+				     elems[i].exp_val.l_val);
+		} else {
+			zassert_false(!bt_mesh_str2ul(elems[i].str, &test_val),
+				      "Conversion of \"%s\" did not return expected err",
+				      elems[i].str);
+		}
+	}
+}
+
+bool compare_float(double x, double y, double epsilon)
+{
+	double temp = x - y;
+
+	temp = temp < 0 ? -temp : temp;
+	return (temp < epsilon);
+}
+
+static void test_str2dbl(void)
+{
+	double test_val = 0;
+
+	const struct input elems[] = {
+		/* Test illegal boundary values */
+#if __SIZEOF_LONG__ == 4
+		{ .is_valid = false, .str = "-2147483649" },
+		{ .is_valid = false, .str = "2147483648" },
+#endif
+		/* Test illegal huge input */
+		{ .is_valid = false, .str = "4294967295000000000000.1" },
+		/* Test corrupt input */
+		{ .is_valid = false, .str = "Corrupt" },
+		{ .is_valid = false, .str = "1234ac" },
+		{ .is_valid = false, .str = "-1234ac" },
+		{ .is_valid = false, .str = "321.-123" },
+		{ .is_valid = false, .str = "." },
+		/* Test legal boundary values */
+		{ .is_valid = true, .str = "2147483647", .exp_val.dbl_val.val = 2147483647,
+		  .exp_val.dbl_val.eps = 0.1f },
+		{ .is_valid = true, .str = "-2147483648", .exp_val.dbl_val.val = -2147483648,
+		  .exp_val.dbl_val.eps = 0.1f },
+		/* Test precision boundary values */
+		{ .is_valid = true, .str = "0.999999999", .exp_val.dbl_val.val = 0.999999999,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = "-0.999999999", .exp_val.dbl_val.val = -0.999999999,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		/* Test precision overflow */
+		{ .is_valid = true, .str = "-0.9999999995", .exp_val.dbl_val.val = -0.999999999,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		/* Test input corner cases */
+		{ .is_valid = true, .str = "-.123", .exp_val.dbl_val.val = -0.123,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = ".123", .exp_val.dbl_val.val = 0.123,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = "00.000012", .exp_val.dbl_val.val = 0.000012,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = "58754.", .exp_val.dbl_val.val = 58754,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		/* Test whitespace correction */
+		{ .is_valid = true, .str = " 0.999999999 ", .exp_val.dbl_val.val = 0.999999999,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = " -0.999999999  ", .exp_val.dbl_val.val = -0.999999999,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = "  -.123 ", .exp_val.dbl_val.val = -0.123,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = true, .str = " 58754.123    ", .exp_val.dbl_val.val = 58754.123,
+		  .exp_val.dbl_val.eps = 0.000000001f },
+		{ .is_valid = false, .str = "         " },
```

Comment:
You're conveniently enough not testing the class of things like `"-12 . 3"` and `".      4"`, with spaces next to the decimal point. These would parse and ignore the internal whitespace. Not sure if that's on purpose.

Reply:
(resolved in person) It would not return an unexpexted value, which is acceptable